pac4j-core/src/main/java/org/pac4j/core/matching/HeaderMatcher.java

```diff
@@ -33,7 +32,9 @@ public boolean matches(final WebContext context) {
 
         final Optional<String> headerValue = context.getRequestHeader(this.headerName);
         final boolean headerNull = expectedValue == null && !headerValue.isPresent();
-        final boolean headerMatches = headerValue.isPresent() && pattern != null && pattern.matcher(headerValue.get()).matches();
+        final boolean headerMatches = headerValue.flatMap(
+            hv -> Optional.ofNullable(pattern).map(p -> p.matcher(hv).matches())
+        ).orElse(false);
```

Comment:
it is strange to see all those code to handle null below (`headerNull`) if we are using optional...